http-client-spi/src/main/java/software/amazon/awssdk/http/HttpMetric.java

```diff
@@ -104,6 +105,19 @@
     public static final SdkMetric<Integer> HTTP_STATUS_CODE =
         metric("HttpStatusCode", Integer.class, MetricLevel.TRACE);
 
+    /**
+     * The time taken to acquire a channel from the channel pool.
+     *
+     * <p>For HTTP/1 operations, a channel is equivalent to a TCP connection. For HTTP/2 operations, a channel is equivalent to
+     * an HTTP/2 stream channel. For both protocols, the time to acquire a new channel may or may not involve establishing a new
+     * connection, depending on whether an existing channel is available in the pool or not. New connection establishment time may
+     * also include TLS handshake time, if TLS is enabled.
+     *
+     * <p>Note: This metric is currently only supported in 'netty-nio-client'.
```

Comment:
Looking to see how feasible this is to add to Apache's `PoolingHttpClientConnectionManager`.

Reply:
This looks to be doable but non-trivial. Deferring to follow-up PR.